scylla/src/transport/cluster.rs

```diff
@@ -339,8 +344,16 @@ impl ClusterData {
         Self::update_rack_count(&mut datacenters);
 
         let keyspaces = metadata.keyspaces;
+        // Exclude these keyspaces from replica set precomputation
+        let keyspaces_to_exclude = match replication_opts {
+            Some(opts) => opts.keyspaces_to_exclude().to_vec(),
+            None => vec![],
+        };
```

Comment:
Better not to allocate here. Let's just use the borrowed `Vec` there.

```suggestion
        // Exclude these keyspaces from replica set precomputation
        let keyspaces_to_exclude = replication_config.keyspaces_to_exclude();
```

Reply:
Even better, let's allow for both Allowlist and Denylist variant:

```suggestion
        let replicas_precomputation_config = replication_config.replicas_precomputation_config();
```